node/src/service.rs

```diff
@@ -1063,4 +1315,209 @@ where
+pub struct OrchestratorChainSolochainInterface {
+    pub overseer_handle: Handle,
+    pub relay_chain_interface: Arc<dyn RelayChainInterface>,
+}
+
+impl OrchestratorChainSolochainInterface {
+    /// Create a new instance of [`RelayChainInProcessInterface`]
+    pub fn new(
+        overseer_handle: Handle,
+        relay_chain_interface: Arc<dyn RelayChainInterface>,
+    ) -> Self {
+        Self {
+            overseer_handle,
+            relay_chain_interface,
+        }
+    }
+}
+
+#[async_trait::async_trait]
+impl OrchestratorChainInterface for OrchestratorChainSolochainInterface {
+    async fn get_storage_by_key(
+        &self,
+        relay_parent: PHash,
+        key: &[u8],
+    ) -> OrchestratorChainResult<Option<StorageValue>> {
+        self.relay_chain_interface
+            .get_storage_by_key(relay_parent, key)
+            .await
+            .map_err(|e| OrchestratorChainError::Application(Box::new(e)))
+    }
+
+    async fn prove_read(
+        &self,
+        relay_parent: PHash,
+        relevant_keys: &Vec<Vec<u8>>,
+    ) -> OrchestratorChainResult<StorageProof> {
+        self.relay_chain_interface
+            .prove_read(relay_parent, relevant_keys)
+            .await
+            .map_err(|e| OrchestratorChainError::Application(Box::new(e)))
+    }
+
+    fn overseer_handle(&self) -> OrchestratorChainResult<Handle> {
+        Ok(self.overseer_handle.clone())
+    }
+
+    /// Get a stream of import block notifications.
+    async fn import_notification_stream(
+        &self,
+    ) -> OrchestratorChainResult<Pin<Box<dyn Stream<Item = PHeader> + Send>>> {
+        self.relay_chain_interface
+            .import_notification_stream()
+            .await
+            .map_err(|e| OrchestratorChainError::Application(Box::new(e)))
+    }
+
+    /// Get a stream of new best block notifications.
+    async fn new_best_notification_stream(
+        &self,
+    ) -> OrchestratorChainResult<Pin<Box<dyn Stream<Item = PHeader> + Send>>> {
+        self.relay_chain_interface
+            .new_best_notification_stream()
+            .await
+            .map_err(|e| OrchestratorChainError::Application(Box::new(e)))
+    }
+
+    /// Get a stream of finality notifications.
+    async fn finality_notification_stream(
+        &self,
+    ) -> OrchestratorChainResult<Pin<Box<dyn Stream<Item = PHeader> + Send>>> {
+        self.relay_chain_interface
+            .finality_notification_stream()
+            .await
+            .map_err(|e| OrchestratorChainError::Application(Box::new(e)))
+    }
+
+    async fn genesis_data(
+        &self,
+        relay_parent: PHash,
+        para_id: ParaId,
+    ) -> OrchestratorChainResult<Option<ContainerChainGenesisData>> {
+        let res: Option<ContainerChainGenesisData> = call_remote_runtime_function(
+            &self.relay_chain_interface,
+            "RegistrarApi_genesis_data",
+            relay_parent,
+            &para_id,
+        )
+        .await
+        .unwrap();
```

Comment:
Can you remove this unwrap?

Reply:
Probably yes for all of them, good catch

Reply:
*fixed*